plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/views/constraintview/ConstraintViewController.java

```diff
@@ -259,14 +212,47 @@ private void checkForRefresh() {
 				viewer.addNoFeatureModelItem();
 			}
 		}
+	}
+
+	/**
+	 * returns a set of constraints if a explanation is available else null
+	 *
+	 */
+	@SuppressWarnings("rawtypes")
+	public List<ConstraintColorPair<IConstraint, Color>> getExplanationConstraints() {
+		if (FeatureModelUtil.getActiveFMEditor() != null) {
+			final FeatureModelEditor fme = FeatureModelUtil.getActiveFMEditor();
+			if (fme.diagramEditor.getActiveExplanation() != null) {
+				final Explanation explanation = (Explanation) fme.diagramEditor.getActiveExplanation();
+				viewer.removeAll();
+				final List<ConstraintColorPair<IConstraint, Color>> constraintList = new ArrayList<ConstraintColorPair<IConstraint, Color>>();
+				for (final Object rea : explanation.getReasons()) {
```

Comment:
More descriptive names (Object rea)

Reply:
Maybe change fmr and fme to fMR and fME for better readability

Reply:
I think lower case is fine for three letter long abbreviations.